test/support/src/vfs_helpers.cc

```diff
@@ -66,6 +66,9 @@ std::vector<std::unique_ptr<SupportedFs>> vfs_test_get_fs_vec() {
       &supports_gcs,
       &supports_rest_s3);
 
+  // some codes uses fs_vec[0] as the default, default to native filesystem
+  fs_vec.emplace_back(std::make_unique<SupportedFsLocal>());
+
```

Comment:
This will fix #4956, awesome!

Reply:
I reverted this because it was not doing the correct thing with `--vfs=rest-s3`
8080705
sorry :(